drivers/eeprom/eeprom_i2c.c

```diff
@@ -50,7 +53,12 @@ static inline void fill_target_address(uint8_t *buffer, const void *addr) {
     }
 }
 
-void eeprom_driver_init(void) { i2c_init(); }
+void eeprom_driver_init(void) {
+    i2c_init();
+#if defined(EXTERNAL_EEPROM_WP_PIN)
+    setPinInputHigh(EXTERNAL_EEPROM_WP_PIN);
```

Comment:
Any particular reason why this is setting the pin as input, rather than `writePin(___, 1)`?

Reply:
Agreed.  Unless there is a specific reason that it's needed, should avoid setting the pin repeatedly.

Reply:
Well, it does need setting repeatedly, it just controls whether or not WP is set on a per-write basis. Setting it to input with a pullup will work, I was just wondering the rationale, considering it's unlikely the WP pin is going to sink any sort of current larger than a microamp or more. Clarity-wise, seeing `writePin(___, 0)` and `writePin(___, 1)` is far more maintainable than trying to decipher why a pin is set to input half of the time.

Reply:
@drashna 
The whole point of the existence of the WP pin is to set it repeatedly, i.e. to have it be zero while writing, and then going back to 1(protected) when idle, so when the keyboard is unplugged, assuming it wasn't writing the eeprom at the time, while the microcontroller is dieing, it doesn't accidentally start a write transaction to the eeprom, which can operate at lower voltages than the microcontroller. Now clearly there are other microcontroller features that can prevent this, such as brownout detector circuits, and such, bit this is one of the intents of the WP pin feature provided by some I2C eeprom manufacturers, I read this in an appnote.

@tzarc 
I'm setting it to inputHigh, rather than outputHigh, because on AVR there's at least 2 bits that have to go bad while the microcontroller is losing power, for it to switch to an incorrect value: the DDRx and the PORTx bit. However now I think I may need to do writePin(___, 1); followed by setPinInputHigh() to achieve the same on other platforms, since it's only on AVR that the PORTx register is overloaded to also control pull-up resistors. I made this change, let me know if you're okay with it. Also added a single line comment.

A secondary reason it's also inputHigh is cause the user might forget to add an external pull-up resistor. This doesn't fix the power-up case, but it does at least improve somewhat the power-down case.